elasticsearch/client.sls

```diff
@@ -0,0 +1,11 @@
+{%- from "elasticsearch/map.jinja" import client with context %}
+{%- if client.get('enabled', False) %}
+
+/etc/salt/minion.d/_elasticsearch.conf:
+  file.managed:
+  - source: salt://elasticsearch/files/_elasticsearch.conf
+  - template: jinja
+  - user: root
+  - group: root
```

Comment:
Yes it may. It is what we are doing for Grafana. And the salt state for Elasticsearch is using the configuration from minion to access the server.